Avoid attaching security groups to NLB in unsupported regions and fix integration tests

hanwen-pcluste opened the pull request:
See commits descriptions for details


### Description of changes
* Tests in the unsupported regions passed

### Tests
* Describe the automated and/or manual tests executed to validate the patch.
* Describe the added/modified tests.

### References
* Link to impacted open issues.
* Link to related PRs in other packages (i.e. cookbook, node).
* Link to documentation useful to understand the changes.

### Checklist
- Make sure you are pointing to **the right branch**.
- If you're creating a patch for a branch other than `develop` add the branch name as prefix in the PR title (e\.g\. `[release-3.6]`).
- Check all commits' messages are clear, describing what and why vs how.
- Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

hanwen-cluster committed:
[integ-tests] Fix AD integration test in iso regions

iso regions don't support tagging certificates. Therefore, the condition in IAM policy is too restrictive in iso reginos

Signed-off-by: Hanwen <hanwenli@amazon.com>

hanwen-cluster committed:
[integ-tests] Fix AD integration tests for regions with FSx OpenZFS and Ontap

Signed-off-by: Hanwen <hanwenli@amazon.com>

hanwen-cluster committed:
Avoid attach security groups to NLB in unsupported regions

NLB with security groups is a new feature added in 2023. However, the feature is not supported in ADC

Signed-off-by: Hanwen <hanwenli@amazon.com>

gmarciani reviewed (changes requested):
See comments in https://github.com/aws/aws-parallelcluster/pull/6447#pullrequestreview-2343477585

hanwen-cluster committed:
[integ-tests] Skip test_shared_home to in regions that do not support FSxOntap and FSxOpenZFS

Prior to this commit, the logic was wrong and didn't skip Ontap and OpenZFS if Lustre is supported

Signed-off-by: Hanwen <hanwenli@amazon.com>

hanwen-pcluste committed:
Merge branch 'release-3.11' into release-3.11oct2

gmarciani commented on cloudformation/ad/ad-integration.yaml:
```diff
@@ -708,9 +708,6 @@ Resources:
                   - iam:UploadServerCertificate
                   - iam:TagServerCertificate
                 Resource: !Sub arn:${AWS::Partition}:iam::${AWS::AccountId}:server-certificate/*
-                Condition:
```
Why are we removing this condition?

hanwen-pcluste replied:
In commit descriptions: iso regions don't support tagging certificates. Therefore, the condition in IAM policy is too restrictive in iso reginos

gmarciani replied:
Then the restriction should be removed only if in ISO regions. Let's do this in a follow up PR

hanwen-pcluste committed:
Merge branch 'release-3.11' into release-3.11oct2
